### PR TITLE
'Bring your own vnet' for admin offer

### PR DESCRIPTION
**Pull request by zhengchang907**
Changes:

* UX: group networking related configurations under **'Networking'** tab.
* Disable DNS configuration when deploying to an existing VNET.
* Not to create NSG when deploying to an existing VNET.
* Use **'StorageV2'** Storage account to enable private endpoint, then mount file share with private IP address to avoid networking issues

Testing:

Manually tested with private offer, with&without SSL using new&existing VNET

Stay as draft because we are collecting feedbacks from partners

### CODE REVIEW

**Comment by edburns** [comment]
Review notes from 2022-05-18 meeting
- On the **Networking** tab, make it so the VNET control is hidden behind a show/hide yes/no radio button.  Set to No by default.  Text "Customize the virtual network for this deployment".  Tooltip, "If yes, you can customize the virtual network into which the offer will deploy the WebLogic Server VM.  If no, the offer will create a new virtual network."

File: weblogic-azure-vm/arm-oraclelinux-wls-admin/src/main/arm/mainTemplate.json [hunk 1]
```diff
@@ -432,6 +432,54 @@
...
+            "new",
+            "existing"
+         ],
+         "metadata": {
+            "description": "Specify whether to create a new or existing virtual network for the VM."
+         }
+      },
+      "virtualNetworkName": {
+         "type": "string",
+         "defaultValue": "jbosseap-vnet",
```
Comment (edburns) [inline weblogic-azure-vm/arm-oraclelinux-wls-admin/src/main/arm/mainTemplate.json:448]: Should not be jbosseap.  Should be **wls-vnet**

File: weblogic-azure-vm/arm-oraclelinux-wls-admin/src/main/arm/mainTemplate.json [hunk 2]
```diff
@@ -432,6 +432,54 @@
...
+            "description": "Name of the existing or new Subnet"
+         }
+      },
+      "subnetPrefix": {
+         "type": "string",
+         "defaultValue": "10.0.0.0/24",
+         "metadata": {
+            "description": "Address prefix of the subnet"
+         }
+      },
```
Comment (edburns) [inline weblogic-azure-vm/arm-oraclelinux-wls-admin/src/main/arm/mainTemplate.json:482]: Are we sure these prefixes are as big as they can be by default?

File: weblogic-azure-vm/arm-oraclelinux-wls-admin/src/main/arm/nestedtemplates/adminTemplate.json [hunk 3]
```diff
@@ -133,6 +133,54 @@
...
+         }
+      },
+      "virtualNetworkName": {
+         "type": "string",
+         "defaultValue": "jbosseap-vnet",
+         "metadata": {
+            "description": "Name of the existing or new VNET"
+         }
+      },
+      "virtualNetworkResourceGroupName": {
```
Comment (edburns) [inline weblogic-azure-vm/arm-oraclelinux-wls-admin/src/main/arm/nestedtemplates/adminTemplate.json:154]: This should be **wls-vnet** not JBoss.

File: weblogic-azure-vm/arm-oraclelinux-wls-admin/src/main/arm/nestedtemplates/adminTemplate.json [hunk 4]
```diff
@@ -133,6 +133,54 @@
...
+         "type": "string",
+         "defaultValue": "wls-subnet",
+         "metadata": {
+            "description": "Name of the existing or new Subnet"
+         }
+      },
+      "subnetPrefix": {
+         "type": "string",
+         "defaultValue": "10.0.0.0/24",
+         "metadata": {
```
Comment (edburns) [inline weblogic-azure-vm/arm-oraclelinux-wls-admin/src/main/arm/nestedtemplates/adminTemplate.json:180]: Are these CIDRs as big as they can be?

File: weblogic-azure-vm/arm-oraclelinux-wls-admin/src/main/arm/nestedtemplates/adminTemplate.json [hunk 5]
```diff
@@ -153,7 +201,7 @@
       }
    },
    "variables": {
-      "const_addressPrefix": "10.0.0.0/16",
+      "const_addressPrefix": "[parameters('addressPrefixes')]",
       "const_hyphen": "-",
       "const_imageOffer": "[concat('weblogic',variables('const_hyphen'), split(variables('name_linuxImageOfferSKU'),variables('const_hyphen'))[1],variables('const_hyphen'),split(variables('name_linuxImageOfferSKU'),variables('const_hyphen'))[2],variables('const_hyphen'),split(variables('name_linuxImageOfferSKU'),variables('const_hyphen'))[3],if(parameters('usePreviewImage'),'-preview',''))]",
       "const_imagePublisher": "oracle",
```
Comment (edburns) [inline weblogic-azure-vm/arm-oraclelinux-wls-admin/src/main/arm/nestedtemplates/adminTemplate.json:207]: Same comment about CIDR.

File: weblogic-azure-vm/arm-oraclelinux-wls-admin/src/main/arm/nestedtemplates/adminTemplate.json [hunk 6]
```diff
@@ -171,22 +219,25 @@
       "const_publicIPAddressType": "Dynamic",
       "const_requiredPortrange": ",65200-65535,5556",
       "const_storageAccountType": "Standard_LRS",
-      "const_subnetPrefix": "10.0.0.0/24",
+      "const_subnetPrefix": "[parameters('subnetPrefix')]",
       "const_vmSize": "[parameters('vmSizeSelect')]",
       "const_wlsHome": "/u01/app/wls/install/oracle/middleware/oracle_home",
       "const_mountPointPath": "[concat('/mnt/', variables('name_share'))]",
       "name_linuxImageOfferSKU": "[first(split(parameters('skuUrnVersion'), ';'))]",
```
Comment (edburns) [inline weblogic-azure-vm/arm-oraclelinux-wls-admin/src/main/arm/nestedtemplates/adminTemplate.json:226]: Same story about CIDR.

File: weblogic-azure-vm/arm-oraclelinux-wls-admin/src/main/arm/nestedtemplates/adminTemplate.json [hunk 7]
```diff
@@ -213,6 +264,7 @@
       {
          "type": "Microsoft.Network/networkSecurityGroups",
          "apiVersion": "${azure.apiVersion}",
+         "condition": "[equals(parameters('virtualNetworkNewOrExisting'), 'new')]",
          "name": "[variables('name_networkSecurityGroup')]",
```
Comment (edburns) [inline weblogic-azure-vm/arm-oraclelinux-wls-admin/src/main/arm/nestedtemplates/adminTemplate.json:268]: See comment in conversation tab about NSG.

File: weblogic-azure-vm/arm-oraclelinux-wls-admin/src/main/arm/nestedtemplates/adminTemplateForCustomSSL.json [hunk 8]
```diff
@@ -183,6 +183,54 @@
...
+         "defaultValue": [
+            "10.0.0.0/16"
+         ],
+         "metadata": {
+            "description": "Address prefix of the VNET."
+         }
+      },
+      "subnetName": {
+         "type": "string",
+         "defaultValue": "wls-subnet",
```
Comment (edburns) [inline weblogic-azure-vm/arm-oraclelinux-wls-admin/src/main/arm/nestedtemplates/adminTemplateForCustomSSL.json:222]: CIDR comment

File: weblogic-azure-vm/arm-oraclelinux-wls-admin/src/main/arm/nestedtemplates/adminTemplateForCustomSSL.json [hunk 9]
```diff
@@ -183,6 +183,54 @@
...
+         "metadata": {
+            "description": "Name of the existing or new Subnet"
+         }
+      },
+      "subnetPrefix": {
+         "type": "string",
+         "defaultValue": "10.0.0.0/24",
+         "metadata": {
+            "description": "Address prefix of the subnet"
+         }
```
Comment (edburns) [inline weblogic-azure-vm/arm-oraclelinux-wls-admin/src/main/arm/nestedtemplates/adminTemplateForCustomSSL.json:232]: CIDR comment

File: weblogic-azure-vm/arm-oraclelinux-wls-admin/src/main/arm/nestedtemplates/adminTemplateForCustomSSL.json [hunk 10]
```diff
@@ -210,7 +258,7 @@
       }
    },
    "variables": {
-      "const_addressPrefix": "10.0.0.0/16",
+      "const_addressPrefix": "[parameters('addressPrefixes')]",
       "const_hyphen": "-",
```
Comment (edburns) [inline weblogic-azure-vm/arm-oraclelinux-wls-admin/src/main/arm/nestedtemplates/adminTemplateForCustomSSL.json:262]: CIDR comment

File: weblogic-azure-vm/arm-oraclelinux-wls-admin/src/main/arm/nestedtemplates/adminTemplateForCustomSSL.json [hunk 11]
```diff
@@ -229,21 +277,24 @@
       "const_publicIPAddressType": "Dynamic",
       "const_requiredPortrange": ",65200-65535,5556",
       "const_storageAccountType": "Standard_LRS",
-      "const_subnetPrefix": "10.0.0.0/24",
+      "const_subnetPrefix": "[parameters('subnetPrefix')]",
       "const_vmSize": "[parameters('vmSizeSelect')]",
```
Comment (edburns) [inline weblogic-azure-vm/arm-oraclelinux-wls-admin/src/main/arm/nestedtemplates/adminTemplateForCustomSSL.json:281]: CIDR comment

File: weblogic-azure-vm/arm-oraclelinux-wls-admin/src/main/arm/mainTemplate.json (hunk 1 above)
Comment (zhengchang907) [reply weblogic-azure-vm/arm-oraclelinux-wls-admin/src/main/arm/mainTemplate.json]: Addressed

File: weblogic-azure-vm/arm-oraclelinux-wls-admin/src/main/arm/mainTemplate.json (hunk 2 above)
Comment (zhengchang907) [reply weblogic-azure-vm/arm-oraclelinux-wls-admin/src/main/arm/mainTemplate.json]: Updated the default value to match the constrains in the UX

File: weblogic-azure-vm/arm-oraclelinux-wls-admin/src/main/arm/nestedtemplates/adminTemplate.json (hunk 4 above)
Comment (zhengchang907) [reply weblogic-azure-vm/arm-oraclelinux-wls-admin/src/main/arm/nestedtemplates/adminTemplate.json]: Updated the default value to match the constrains in the UX

**Review by edburns — APPROVED** [review]

**Review by mriccell — APPROVED** [review]

**Pull request merged**